taichi/gui/x11.cpp

```diff
@@ -89,8 +89,28 @@ void GUI::process_event() {
       case ButtonPress:
         set_mouse_pos(ev.xbutton.x, height - ev.xbutton.y - 1);
         mouse_event(MouseEvent{MouseEvent::Type::press, cursor_pos});
-        key_events.push_back(
-            KeyEvent{KeyEvent::Type::press, lookup_button(&ev), cursor_pos});
+        switch (ev.xbutton.button) {
+        case 4: // wheel up
+          key_events.push_back(
+              KeyEvent{KeyEvent::Type::press, "Wheel", cursor_pos, Vector2i{0, +120}});
```

Comment:
Can i ask why using a static value...?

Reply:
IIRC , X11 doesn't give out a concrete delta value within the wheel event struct .
And 120 is a general value on Win32 (see ```WHEEL_DELTA```)and Qt , which is set as 8 multiple of wheel angle in one step. We just reuse it on X11 .

Reply:
> Can i ask why using a static value...?

IIRC , X11 doesn't give out a concrete delta value within the wheel event struct .
And 120 is a general value on Win32 (see WHEEL_DELTA)and Qt , which is set as 8 multiple of wheel angle in one step. We just reuse it on X11 .